src/api-umbrella/proxy/jobs/distributed_rate_limit_puller.lua

```diff
@@ -89,7 +78,10 @@ local function check(premature)
     return
   end
 
-  local ok, err = pcall(do_check)
+  -- here we subtract the lock expiration time by 1ms to prevent
+  -- a race condition with the next timer event.
+  local ok, err = interval_lock('distributed-last-pull', delay - 0.001,
```

Comment:
It looks like we end up subtracting 2ms to handle the race condition, since this subtraction is happening here and also inside the `interval_lock` function. Not a huge deal, but I think it might make the most sense to only subtract 1ms inside `interval_lock` and shift the comment explaining the 1ms funkiness into the `interval_lock` function. That way the caller doesn't really have to be aware of this issue.